### PR TITLE
Check for non-blocked status when determining isMergeable

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details
Adds a check for the `mergeable_state` of a PR to make sure it is not `BLOCKED`. This is the state that a PR can get in with unverified commits, messing up the deploy process.

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/190589

### Tests
1. I verified that PRs that are not able to be auto-merged if they are not in the [`mergeable_state` of `"clean", "has_hooks", "unknown", "unstable"`](https://github.com/pascalgn/automerge-action/blob/main/lib/merge.js#L5), so this will prevent auto-merge from silently failing in the `BLOCKED` state

### QA Steps
N/A

### CODE REVIEW

File: .github/actions/isPullRequestMergeable/index.js [hunk 1]
```diff
@@ -37,8 +37,15 @@ const run = function () {
...
+                if (_.isNull(data.mergeable_state)) {
+                    console.log('Pull request mergeable_state is not yet resolved...');
+                    retryCount++;
+                    return;
+                }
+
                 mergeabilityResolved = true;
-                isMergeable = data.mergeable;
+                console.log(`Merge information for #${pullRequestNumber} - mergeable: ${data.mergeable}, mergeable_state ${data.mergeable_state}`);
+                isMergeable = data.mergeable && data.mergeable_state !== 'BLOCKED';
```
Comment (Jag96) [inline .github/actions/isPullRequestMergeable/index.js:48]: It looks like in the images the value is lowercase, should we uppercase the `data.mergeable_state` value to ensure the comparison is exact?

File: .github/actions/isPullRequestMergeable/index.js [hunk 2]
```diff
@@ -37,8 +37,15 @@ const run = function () {
...
 
+                if (_.isNull(data.mergeable_state)) {
+                    console.log('Pull request mergeable_state is not yet resolved...');
+                    retryCount++;
+                    return;
+                }
+
                 mergeabilityResolved = true;
-                isMergeable = data.mergeable;
+                console.log(`Merge information for #${pullRequestNumber} - mergeable: ${data.mergeable}, mergeable_state ${data.mergeable_state}`);
```
Comment (Jag96) [inline .github/actions/isPullRequestMergeable/index.js:47]: nab - `mergeable_state: ${data.mergeable_state}` for consistency

File: .github/actions/isPullRequestMergeable/isPullRequestMergeable.js [hunk 3]
```diff
@@ -27,8 +27,15 @@ const run = function () {
                     return;
                 }
 
+                if (_.isNull(data.mergeable_state)) {
```
Comment (roryabraham) [inline .github/actions/isPullRequestMergeable/isPullRequestMergeable.js:30]:
```suggestion
                if (_.isEmpty(data.mergeable_state)) {
```

File: .github/actions/isPullRequestMergeable/isPullRequestMergeable.js [hunk 4]
```diff
@@ -27,8 +27,15 @@ const run = function () {
...
+                    console.log('Pull request mergeable_state is not yet resolved...');
+                    retryCount++;
+                    return;
+                }
+
                 mergeabilityResolved = true;
-                isMergeable = data.mergeable;
+                console.log(`Merge information for #${pullRequestNumber} - mergeable: ${data.mergeable}, mergeable_state ${data.mergeable_state}`);
+                isMergeable = data.mergeable && data.mergeable_state !== 'BLOCKED';
```
Comment (roryabraham) [inline .github/actions/isPullRequestMergeable/isPullRequestMergeable.js:38]:
```suggestion
                isMergeable = data.mergeable && data.mergeable_state.toUpperCase() !== 'BLOCKED';
```

**Review by roryabraham — DISMISSED** [review]

File: .github/actions/isPullRequestMergeable/index.js (hunk 1 above)
Comment (AndrewGable) [reply .github/actions/isPullRequestMergeable/index.js:48]: Updated with @roryabraham's suggestion which should handle this

**Review by roryabraham — APPROVED** [review]

**Review by roryabraham — CHANGES_REQUESTED** [review]
Code changes look good, but I think we might need to update the unit tests now

**Comment by AndrewGable** [comment]
Sorry looks like a jest test failed, looking into it now.

**Comment by AndrewGable** [comment]
Jest tests should be passing now, just waiting on E2E for the ✅

**Review by roryabraham — APPROVED** [review]

**Review by Jag96 — APPROVED** [review]

**Pull request merged**